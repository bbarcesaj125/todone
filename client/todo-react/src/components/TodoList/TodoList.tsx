import React from "react";
import TodoSingle from "../TodoSingle";

interface Props {
  items: Item[];
  toggleItem: ToggleItem;
}
const TodoList: React.FC<Props> = ({ items, toggleItem }) => {
  return (
    <ul>
      {items.map((item) => (
        <TodoSingle key={item.title} item={item} toggleItem={toggleItem} />
      ))}
    </ul>
  );
};

export default TodoList;
