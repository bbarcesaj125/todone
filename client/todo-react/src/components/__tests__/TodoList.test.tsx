import React from "react";
import { render, screen } from "@testing-library/react";
import TodoList from "../TodoList";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import todoData from "../../todos.json";

const initialItems: Item[] = todoData;

it("TodoList displays a list of tasks", () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/"]}>
      <Routes>
        <Route path="/" element={<TodoList items={initialItems} />} />
      </Routes>
    </MemoryRouter>
  );

  initialItems.forEach((item) => {
    const taskTitle = screen.getByText(item.title);
    expect(taskTitle).toBeInTheDocument;
  });
  expect(asFragment()).toMatchSnapshot();
});
