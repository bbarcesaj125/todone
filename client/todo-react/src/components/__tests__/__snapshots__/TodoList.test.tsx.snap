// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList displays a list of tasks 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <input
        checked=""
        type="checkbox"
      />
      <a
        href="/task/Reading"
      >
        <label
          class="sc-bczRLJ cFTAWH"
          title="Reading"
        >
          Reading
        </label>
      </a>
    </li>
    <li>
      <input
        checked=""
        type="checkbox"
      />
      <a
        href="/task/Shopping"
      >
        <label
          class="sc-bczRLJ cFTAWH"
          title="Shopping"
        >
          Shopping
        </label>
      </a>
    </li>
    <li>
      <input
        type="checkbox"
      />
      <a
        href="/task/Github"
      >
        <label
          class="sc-bczRLJ cKhOHf"
          title="Github"
        >
          Github
        </label>
      </a>
    </li>
    <li>
      <input
        type="checkbox"
      />
      <a
        href="/task/Writing"
      >
        <label
          class="sc-bczRLJ cKhOHf"
          title="Writing"
        >
          Writing
        </label>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`TodoSingle should be checked for the task 'Reading' 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <input
        checked=""
        type="checkbox"
      />
      <a
        href="/task/Reading"
      >
        <label
          class="sc-bczRLJ cFTAWH"
          title="Reading"
        >
          Reading
        </label>
      </a>
    </li>
    <li>
      <input
        checked=""
        type="checkbox"
      />
      <a
        href="/task/Shopping"
      >
        <label
          class="sc-bczRLJ cFTAWH"
          title="Shopping"
        >
          Shopping
        </label>
      </a>
    </li>
    <li>
      <input
        type="checkbox"
      />
      <a
        href="/task/Github"
      >
        <label
          class="sc-bczRLJ cKhOHf"
          title="Github"
        >
          Github
        </label>
      </a>
    </li>
    <li>
      <input
        type="checkbox"
      />
      <a
        href="/task/Writing"
      >
        <label
          class="sc-bczRLJ cKhOHf"
          title="Writing"
        >
          Writing
        </label>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
